src/main/java/io/th0rgal/oraxen/mechanics/provided/combat/lifeleech/LifeLeechMechanicListener.java

```diff
@@ -37,8 +37,17 @@ public void onCall(EntityDamageByEntityEvent event) {
         double health = damager.getHealth() + mechanic.getAmount();
         double maxHealth = damager.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue();
 
-        damager.setHealth(Math.min(health, maxHealth));
         if (event.getEntity() instanceof LivingEntity damaged) {
+            if(damaged.hasMetadata("NPC")){
+                // Its a NPC so just cancel the event and return
+                event.setCancelled(true);
+                return;
+            }
```

Comment:
this feels like it can be done in a better way
Preferably use Protectionlib.canInteract() method to check
Same goes for most of these mechanics

Reply:
but that don't fix incompatibility with citizens i guess? don't know much about ProtectionLib but according to citizens wiki they add a metadata called "NPC" to entity's they spawn
![image](https://user-images.githubusercontent.com/63419731/233620712-950fa90c-80c3-41b8-a447-4d5488e0515d.png)

Reply:
would rather use actual api methods over this as NPC might, even if rare, be used elsewhere
plus it isnt actually obvious here what ur checking

tldr make a softdepend on citizens and use api to check it. otherwise ill get around to this whenever i get time to it

Reply:
yeah sure, but to explain what I am checking. I am basically checking if the "damaged" entity is a npc or not if its an npc then don't do anything just ignore else do the normal functionality of the Lifeleech mechanic

Reply:
you are checking if it is a Citizens NPC
checking the metadata will probably fix it but it isnt very descriptive of what you are doing
atleast add comments that it is checking if it is a citizens npc

cancelling the event is also not something you should do here
the listeners job isnt to do this and will probably cause issues with other plugins
simply returning is enough

Reply:
alright so, i will do these changes

Reply:
did the changes, do you still want me to use the actual api to check or just the metadata check is enough?

Reply:
also i think its better to add a method called #isNpc(Entity) in ProtectionLib so we can add support to multiple npc plugins other than citizens